cmd/minishift/cmd/profile/profile_set.go

```diff
@@ -40,22 +36,44 @@ var profileSetCmd = &cobra.Command{
 func runProfile(cmd *cobra.Command, args []string) {
 	validateArgs(args)
 	profileName := args[0]
+	lastActiveProfile := profileActions.GetActiveProfile()
 
 	if !cmdUtil.IsValidProfile(profileName) {
+		if cmdUtil.GetVMStatus(lastActiveProfile) != "Does Not Exist" {
+			err := cmdUtil.DeleteOcContext(lastActiveProfile)
+			if err != nil {
+				if glog.V(2) {
+					fmt.Println(fmt.Sprintf("%s", err.Error()))
+				}
+				fmt.Println(fmt.Sprintf("Could not delete context of previous profile: %s", lastActiveProfile))
+			}
+		}
 		err := profileActions.SetActiveProfile(profileName)
 		if err != nil {
 			atexit.ExitWithMessage(1, err.Error())
 		}
 	} else {
-		err := cmdUtil.SetOcContext(profileName)
-		if err != nil {
-			if glog.V(2) {
-				fmt.Println(fmt.Sprintf("%s", err.Error()))
+		if cmdUtil.GetVMStatus(lastActiveProfile) != "Does Not Exist" {
```

Comment:
The state strings should really be constants. Let's try to define them at one place only